lang/java/avro/src/main/java/org/apache/avro/LogicalTypes.java

```diff
@@ -30,6 +30,10 @@
 
   public interface LogicalTypeFactory {
     LogicalType fromSchema(Schema schema);
+
+    default String getTypeName() {
```

Comment:
I know this `default` is here for backwards-compatibility but I'm not sure about the value. If I create a factory - "OffsetDateTimeFactory" - I wouldn't want that to be the logicalType name. In other words, what do you think about the default implementation being:
`default String getTypeName() {
   throw new UnsupportedOperationException();
}
`
This way it is required by any new LogicalTypeFactory to implement it?

Reply:
Good point. 
I propose to use just `String getTypeName()` for this. Your proposition is more for abstract class.